components/script/dom/offscreencanvas.rs

```diff
@@ -121,8 +120,15 @@ impl OffscreenCanvas {
         Some(context)
     }
 
-    pub(crate) fn is_valid(&self) -> bool {
-        self.Width() != 0 && self.Height() != 0
+    pub(crate) fn is_paintable(&self) -> bool {
+        let size = self.get_size();
+
+        let paintable = match *self.context.borrow() {
+            Some(OffscreenCanvasContext::OffscreenContext2d(ref context)) => context.is_paintable(),
+            _ => false,
+        };
+
+        (paintable && !size.is_empty()) || is_supported_canvas_size(size)
```

Comment:
Could you explain why OffScreenCanvas needs to check more than `paintable`?

Reply:
You compared with outdated HTMLCanvasElement.rs (in the latest version they have similar check)
https://github.com/servo/servo/pull/36445/files#diff-7483a0f0ae501ab9da1432c28321c22f39647521072498f773903f360b026ce3R388

Reply:
I'm still not following why some canvas types need extra `is_supported_canvas_size` checks and some do not.

In an early message you mentioned that WebGL needs this check as a fixup, in which case I believe it should be moved to WebGL related types.

If it's general, then could you elaborate the difference between the paintability of a DrawTarget and that of a canvas (be it HTMLCanvas or offscreen)?

Reply:
HTMLCanvasElement "paintable" (top level)
   - ContextCanvas2D DrawTarget "paintable" (low level)
   - WebGL/WebGPU DrawingBuffer "paintable" (low level)
 

There are some places in code there we could/must check possibility to snapshot image data OR draw from other canvas and current proposal was to up checking of "paintability" to top level instead of waiting until it will be done at context level (but it will be more CPU resourceble because we can do validation twice, but could help on some edge cases).

NOTE:
"Paintable" on low level (per context) will be validated against draw target size limitations + context lost status (for GPU accelerated rendering - not yet supported on 100%)